lib/fxa-geodb.js

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var DEFAULTS = require('./defaults');
+var ERRORS = require('./errors');
+var maxmind = require('maxmind');
+var Promise = require('bluebird');
+
+module.exports = function (options) {
+  'use strict';
+
+  options = options || {};
+  var dbPath = options.dbPath || DEFAULTS.DB_PATH;
+
+  var dbLookup;
+  // we quit if the db did not load for some reason
+  try {
+    dbLookup = maxmind.open(dbPath);
+  } catch (err) {
+    // if it failed with primary database
+    // then quit with error
+    throw ERRORS.UNABLE_TO_OPEN_FILE;
```

Comment:
Just to add to @vladikoff's comment in https://github.com/mozilla/fxa-auth-server/pull/1334/files#r71040754, failure to get geolocation information should not be treated as a fatal error. The notification **must** be sent, and deal with missing location data in a way that makes the most sense to the user (which is what will have to happen anyways, since a given IP address is not guaranteed to have location information, or may not even even be accurate).

However, we should work on a way to scream loudly that we are not getting geolocation "hits", whether the cause is a missing or corrupt or invalid data file, or an error in this module, or un-resolvable IP addresses.